doc/user/content/sql/create-source/webhook.md

```diff
@@ -22,13 +22,18 @@ Webhook sources expose a [public URL](#webhook-url) that allows your application
 
 {{< diagram "webhook-check-option.svg" >}}
 
+### `with_options`
+
+{{< diagram "with-options-retain-history.svg" >}}
+
 Field                            | Use
 ---------------------------------|--------------------------
   _src_name_                     | The name for the source
  **IN CLUSTER** _cluster_name_   | The [cluster](/sql/create-cluster) to maintain this source.
  **INCLUDE HEADER**              | Map a header value from a request into a column.
  **INCLUDE HEADERS**             | Include a column named `'headers'` of type `map[text => text]` containing the headers of the request.
  **CHECK**                       | Specify a boolean expression that is used to validate each request received by the source.
+ **RETAIN HISTORY FOR** <br>_retention_period_ | ***Private preview.** This option has known performance or stability issues and is under active development.* Duration for which Materialize retains historical data for performing [time travel queries](/transform-data/patterns/time-travel-queries). Accepts positive [interval](/sql/types/interval/) values (e.g. `'1hr'`). Default: `1s`.
```

Comment:
I'm not sure this works for webhooks!

Reply:
Interesting, when I tried to alter my webhook source to set retain history i got the following error
```
Error: internal error: planner should have rejected invalid alter retain history item type
```
I'll remove from the docs for now and it's non-blocking but figured I'd raise since it seems like the error message is unexpected.

Reply:
hey a bug! filed https://github.com/MaterializeInc/materialize/issues/27020